src/components/tx/security/redefine/RedefineBalanceChange/index.tsx

```diff
@@ -0,0 +1,145 @@
+import EthHashInfo from '@/components/common/EthHashInfo'
+import TokenIcon from '@/components/common/TokenIcon'
+import useBalances from '@/hooks/useBalances'
+import useChainId from '@/hooks/useChainId'
+import { useHasFeature } from '@/hooks/useChains'
+import { type RedefineModuleResponse } from '@/services/security/modules/RedefineModule'
+import { sameAddress } from '@/utils/addresses'
+import { FEATURES } from '@/utils/chains'
+import { formatVisualAmount } from '@/utils/formatters'
+import { Box, Chip, Grid, Typography } from '@mui/material'
+import { TokenType } from '@safe-global/safe-gateway-typescript-sdk'
+import { useContext } from 'react'
+import { LoadingLabel } from '../../shared/LoadingLabel'
+import { TransactionSecurityContext } from '../../TransactionSecurityContext'
+import css from './styles.module.css'
+
+const SingleFungibleBalanceChange = ({
+  change,
+  positive = false,
+}: {
+  change: NonNullable<RedefineModuleResponse['balanceChange']>['in' | 'out'][number] & { type: 'ERC20' | 'NATIVE' }
+  positive?: boolean
+}) => {
+  const { balances } = useBalances()
+
+  const logoUri = balances.items.find((item) => {
+    return change.type === 'NATIVE'
+      ? item.tokenInfo.type === TokenType.NATIVE_TOKEN
+      : sameAddress(item.tokenInfo.address, change.address)
+  })?.tokenInfo.logoUri
+  return (
+    <>
+      <Typography>
+        {positive && '+'}
+        {formatVisualAmount(change.amount.value, change.decimals)}
+      </Typography>
+      <TokenIcon size={24} logoUri={logoUri} tokenSymbol={change.symbol} />
+      <Typography fontWeight={700} display="inline">
+        {change.symbol}
+      </Typography>
+      <Chip sx={{ borderRadius: '4px' }} label={change.type} />
+    </>
+  )
+}
+
+const SingleNFTBalanceChange = ({
+  change,
+  positive = false,
+}: {
+  change: NonNullable<RedefineModuleResponse['balanceChange']>['in' | 'out'][number] & { type: 'ERC721' }
+  positive?: boolean
+}) => {
+  const chainId = useChainId()
+
+  return (
+    <>
+      <Typography mr={-1}>{positive ? '+' : '-'}</Typography>
+      {change.symbol ? (
+        <Typography fontWeight={700} display="inline">
+          {change.symbol}
+        </Typography>
+      ) : (
+        <EthHashInfo
+          name={change.symbol ?? 'NFT'}
+          address={change.address}
+          chainId={chainId}
+          showCopyButton={false}
+          showPrefix={false}
+          hasExplorer
+          showAvatar={false}
+          shortAddress
+        />
+      )}
+      <Typography variant="subtitle2" fontWeight={700} className={css.nftId}>
+        #{change.tokenId}
+      </Typography>
+      <Chip sx={{ borderRadius: '4px' }} label="NFT" />
+    </>
+  )
+}
+
+const SingleBalanceChange = ({
```

Comment:
nit: If there is no `MultipleBalanceChanges` component or something similar you could also shorten the name to `BalanceChange` as its already singular.